test/test_sparse_semi_structured.py

```diff
@@ -1174,16 +1174,28 @@ def test_cslt_sparse_mm_search(self, device, dtype):
         alg_id = torch._cslt_sparse_mm_search(A_compressed, B.t())
         assert alg_id in range(torch.backends.cusparselt.get_max_alg_id())
 
+    @inference_dtypes
+    def test_csrc_cslt_sparse_mm_search(self, device, dtype):
+        A = rand_sparse_semi_structured_mask(256, 128, dtype=dtype)
+        A_compressed = torch._cslt_compress(A)
+        B = torch.ones((128, 128), device=device).to(dtype)
+
+        A_compressed = torch._cslt_compress(A)
+        alg_id, _, _, _ = torch._C._cusparselt.mm_search(A_compressed, B.t(), None, None, None, False)
+        assert alg_id in range(torch.backends.cusparselt.get_max_alg_id())
+
     def test_cusparselt_backend(self):
         version = _get_torch_cuda_version()
         assert torch.backends.cusparselt.is_available()
 
         # CUDA 11.8 has cuSPARSELt v0.4.0 support
         if version == (11, 8):
             assert torch.backends.cusparselt.version() == 400
+            assert torch.backends.cusparselt.get_max_alg_id() == 4
```

Comment:
max_alg_id depends on datatype and layouts and is not a constant in all versions including 0.4.0.
If needed, its value should always be retrieved using attribute `CUSPARSELT_MATMUL_ALG_CONFIG_MAX_ID`.

Reply:
gotcha, will remove the global alg_id reference then.